### ISSUE TITLE
Linting Inside of a JS String

### ISSUE THREAD

**Issue by dannyfritz** [liuderchi/ide-css#5]
`ide-css` lets me modify the scopes it can operate in. Is there a way to get it to operate within a template literal of JS? The syntax highlighting already works, just need to know if there is a scope I can tell `ide-css` to look at to make this work.
![atom_2017-10-04_14-49-42](https://user-images.githubusercontent.com/315788/31199218-e9d27fd2-a913-11e7-94ce-68b27aa88da4.png)

**Comment by liuderchi** [comment]
@dannyfritz 

**TL;DR;**
Currently `ide-css` CAN NOT do css linting (or other parsing) in file of JS-grammar because
it operates according to ONLY the *Root Scope* of the file.

---

To get more info you may do `Editor: Log Cursor Scope` command with cursor on css codes;
you will see list of scopes starting from root scope (grammar of the file) to children scopes.

`ide-css` cares ONLY about root scope while syntax highlighting cares about different children scopes line by line.

**Comment by dannyfritz** [comment]
My understanding is that this is currently a limitation of `vscode-css-languageserver-bin`. Is that correct?

**Comment by liuderchi** [comment]
> My understanding is that this is currently a limitation of vscode-css-languageserver-bin. Is that correct?

Yes. Vscode-css-languageserver-bin parses WHOLE file with only css syntax knowledge.
IMO this feature may be an extension for [javascript-typescript-langserver](https://github.com/sourcegraph/javascript-typescript-langserver)
